internal/tiltfile/tiltfile_state.go

```diff
@@ -1478,6 +1479,28 @@ func (s *tiltfileState) translateLocal() ([]model.Manifest, error) {
 
 	return result, nil
 }
+func (s *tiltfileState) warnDuplicateYamlResources(m model.Manifest) {
+	var tempName []string
+	deployTarget := m.K8sTarget()
+	displayNames := deployTarget.DisplayNames
+
+	duplicates := make(map[string]int)
+
+	// depends on the fact that the duplicate resource has formatting of doggos:service:default:core:n where n is nth
+	//occurrence of that resource.
+	for _, name := range displayNames {
+		tempName = strings.Split(name, ":")
+		if len(tempName) > 4 { //if it is > 4 that means there's an n
+			duplicates[tempName[0]]++ // we only care about the name of the resource
+		}
+	}
+
+	for resource, numOccurences := range duplicates {
+		if numOccurences > 1 {
+			s.logger.Warnf("The following YAML Resource has been duplicated: " + resource)
```

Comment:
A. as said in a comment above, this should be "entity" and not "resource"
B. think about whether this gives the user enough information to debug. Does it say what resource the duplicated entities belong to? Why is this a problem, how should the user fix it?
C. (minor) I'm not sure you want one warning for each duplicated entity; consider instead [something like this](https://github.com/tilt-dev/tilt/blob/3f7711b77d4885b5e1ee2026b56f9edec9a0c8ca/internal/tiltfile/live_update.go#L25) where you list all the offenders in a single warning